src/plugins/analysis/cve_lookup/internal/setup_repository.py

```diff
@@ -0,0 +1,254 @@
+        update_cve_summaries()
+        drop_table(table_name='temp_sum')
+
+    drop_table(table_name='temp_feeds')
+
+
+def update_cve_feeds():
+    feeds_to_be_updated = extract_relevant_feeds(from_table='temp_feeds', where_table='cve_table')
+    delete_outdated_feeds(delete_outdated_from='cve_table', use_for_selection='temp_feeds')
+    insert_into(query='insert_cve', table_name='cve_table', input_data=feeds_to_be_updated)
+
+
+def update_cve_summaries():
+    summaries_to_be_updated = extract_relevant_feeds(from_table='temp_sum', where_table='cve_table')
+
+    if exists(table_name='summary_table'):
+        delete_outdated_feeds(delete_outdated_from='summary_table', use_for_selection='temp_sum')
+        delete_outdated_feeds(delete_outdated_from='summary_table', use_for_selection='temp_feeds')
+    else:
+        create(query='create_summary_table', table_name='summary_table')
+
+    delete_outdated_feeds(delete_outdated_from='cve_table', use_for_selection='temp_sum')
+    insert_into(query='insert_summary', table_name='summary_table', input_data=summaries_to_be_updated)
+
+
+def get_years_from_database():
+    return [year for (year,) in DATABASE.select_query(QUERIES['get_years_from_cve'])]
+
+
+def import_cve(cve_extract_path: str, years: namedtuple):
+    if exists(table_name='cve_table'):
+        filtered_years = overlap(requested_years=years, years_in_cve_database=get_years_from_database())
+        year_selection = filtered_years if filtered_years else list(range(years.start_year, years.end_year + 1))
+    else:
+        year_selection = list(range(years.start_year, years.end_year + 1))
+
+    cve_list, summary_list = get_cve_import_content(cve_extraction_path=cve_extract_path, year_selection=year_selection)
+    init_cve_feeds_table(cve_list=cve_list, table_name='cve_table')
+
+    if summary_list:
+        init_cve_summaries_table(summary_list=summary_list, table_name='summary_table')
+
+
+def init_repository(extraction_path: str, specify: int, years: namedtuple):
+    if specify == 0:
+        import_cpe(cpe_extract_path=extraction_path)
+        import_cve(cve_extract_path=extraction_path, years=years)
+    elif specify == 1:
+        import_cpe(cpe_extract_path=extraction_path)
+    elif specify == 2:
+        import_cve(cve_extract_path=extraction_path, years=years)
+
+
+def update_repository(extraction_path: str, specify: int):
+    if specify == 0:
+        update_cpe(extraction_path)
+        update_cve_repository(extraction_path)
+    elif specify == 1:
+        update_cpe(extraction_path)
+    elif specify == 2:
+        update_cve_repository(extraction_path)
+
+
+def setup_argparser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--update', '-u',
+        help='Boolean which specifies if the DATABASE should be updated. Default: False',
+        type=bool,
+        default=False
+    )
+    parser.add_argument(
+        '--specify', '-s',
```

Comment:
#nitpick but maybe _specify_ is a bit misleading?

Reply:
refactored

Reply:
I'd have liked if you had kept the naming consistent (i.e. class Choice but option target, why not class Target). But it's def. better than before